src/lrs.rs

```diff
@@ -468,16 +479,42 @@ impl<CurveImpl: Curve> LrsBase for Lrs<CurveImpl> {
         })
     }
 
+    fn traversals_containing(&self, point: Point) -> Vec<TraversalHandle> {
+        let rtree = self
+            .rtree_data
+            .as_ref()
+            .and_then(|buf| RTreeRef::try_new(buf).ok());
+
+        rtree
+            .map(|tree| {
+                tree.neighbors_coord(&coord! {x: point.x(), y: point.y()}, Some(10), None)
+                    .iter()
+                    .map(|idx| *idx as usize)
+                    .filter(|idx| self.traversals[*idx].curve.bbox().contains(&point))
+                    .map(TraversalHandle)
+                    .collect()
+            })
+            .unwrap_or(
+                self.traversals
+                    .iter()
+                    .enumerate()
+                    .filter(|(_idx, traversal)| traversal.curve.bbox().contains(&point))
+                    .map(|(idx, _traversal)| TraversalHandle(idx))
+                    .collect(),
+            )
+    }
+
     fn lookup_lrms(&self, point: Point) -> Vec<LrmProjection> {
         let mut result: Vec<_> = self
-            .lrms
+            .traversals_containing(point)
             .iter()
-            .enumerate()
-            .flat_map(|(lrm_idx, _lrm)| self.lookup(point, LrmHandle(lrm_idx)))
+            .flat_map(|traversal_handle| self.traversals[traversal_handle.0].lrms.iter())
```

Comment:
do you need the final `.iter()`? If the compiler complains about consuming with `.into_iter()` (which `.flat_map` does I believe), you should be able to only give a reference then.


```suggestion
            .flat_map(|traversal_handle| &self.traversals[traversal_handle.0].lrms)
```